src/Plutarch/Test/QuickCheck.hs

```diff
@@ -268,6 +270,21 @@ haskEquiv' h p =
             (Proxy @Arbitrary)
             arbitrary
 
+{- | Compares evaluated UPLC
+
+ @since 2.0.1
+-}
+uplcEq :: TestableTerm a -> TestableTerm b -> Property
+uplcEq x y = property $ eval x == eval y
+  where
+    eval (TestableTerm t) =
+        case compile (Config{tracingMode = NoTracing}) t of
+            Left err -> error $ show err
+            Right s' ->
+                case evalScript s' of
+                    (Right s, _, _) -> s
+                    (Left err, _, _) -> error $ show err
```

Comment:
Is it necessary to `error` out in this case? We could have the property fail rather than `error` I think.

Reply:
Yeah this should use property fail and have pretty errors